File: internal/content/testdata/bench/go/majority-element-half.go
func majority_element(nums []int) int {
	count := 0
	candidate := 0
	for _, n := range nums {
		if count == 0 {
			candidate = n
		}
		if n == candidate {
			count++
		} else {
			count--
		}
	}
	cnt := 0
	for _, n := range nums {
		if n == candidate {
			cnt++
		}
	}
	if cnt > len(nums)/2 {
		return candidate
	}
	return -1
}

File: internal/content/testdata/bench/csharp/majority-element-half.cs
public class Solution {
    public long majority_element(long[] nums) {
        long count = 0;
        long candidate = 0;
        foreach (long n in nums) {
            if (count == 0) candidate = n;
            count += (n == candidate) ? 1 : -1;
        }
        long cnt = 0;
        foreach (long n in nums) if (n == candidate) cnt++;
        if (cnt > nums.Length / 2) return candidate;
        return -1;
    }
}
